### PR TITLE
Add fMoW dataset loader

### PR DESCRIPTION
**Pull request by arpitJ-dev**
### Summary

The FMoW loader supports the official train and val splits of the approximately 200 GB fMoW-rgb dataset. It searches for files matching `<root>/<split>/<category>/<sequence>/*_rgb.jpg` and sorts the discovered paths to provide deterministic ordering. The category is extracted from the image's directory path and converted into an integer using the loader's fixed 62-category list.

Each JPEG is converted to RGB and returned as a float32 tensor in `[channels, height, width]` format. The corresponding JSON file is found by replacing the image's `.jpg` extension with `.json`. Every metadata bounding box is converted from fMoW's `[x, y, width, height]` format into `[x1, y1, x2, y2]` by calculating `x2 = x + width` and `y2 = y + height`. The result is always reshaped to `N x 4`; therefore, metadata containing no boxes produces a consistent empty tensor with shape `(0, 4)`.

The resulting sample contains `image`, `label`, and `bbox_xyxy`. The accompanying `FMoWDataModule` provides the standard train and validation data loaders. If a transform is supplied, it receives this complete sample after all three values have been constructed. The tests use an identity transform, confirming that the transform path accepts and returns the complete sample without changing it.

The plotting method changes the image from channel-first `[C, H, W]` form to display-ready `[H, W, C]` form. It hides the axes and normally shows the readable category name as the title. If the sample contains a prediction, the predicted category is added on a second title line. Passing `show_titles=False` hides both the label and prediction title, while `suptitle` is handled independently and can still appear above the figure. The plotting method also draws the `bbox_xyxy` annotations.

### Rationale

Automatic downloading is not implemented. Users must explicitly download the dataset's train and val directories under the selected root. The dataset documentation includes the AWS CLI commands for both splits. This avoids unexpectedly initiating a roughly 200 GB transfer.

See #2580.

### Checklist

- [x] I have read the [Contributing docs](https://docs.torchgeo.org/en/stable/user/contributing.html)
- [x] I have read the [AI policy](https://github.com/torchgeo/governance/blob/main/AI-POLICY.md)

- [ ] 🟢 **No AI usage**: written by humans, for humans
- [x] 🟡 **AI-assisted**: AI helped with the coding, but I understand every line
- [ ] 🔴 **AI-generated**: AI did everything, review with caution

### Testing

- `python -m pytest tests/datasets/test_fmow.py tests/datamodules/test_fmow.py -q` (18 passed).
- Ruff formatting and lint checks.
- `ty` static type checks.
- Official hosted-data smoke test.

### Sample plot

fmow_airport_sample

### CODE REVIEW

File: torchgeo/datasets/fmow.py [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+        with open(json_path, encoding='utf-8') as f:
+            metadata = json.load(f)
+
+        for annotation in metadata['bounding_boxes']:
+            x, y, w, h = annotation['box']
+            boxes.append([float(x), float(y), float(x + w), float(y + h)])
+
+        return torch.tensor(boxes, dtype=torch.float32).reshape(-1, 4)
+
+    def plot(
```
Comment (nilsleh) [inline torchgeo/datasets/fmow.py:236]: if there are bounding boxes, can you add plotting of the bounding boxes? Additionally, can you add a sample plot to your PR message to check data loading and plotting

File: torchgeo/datasets/fmow.py [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+
+        sample: Sample = {'image': image, 'label': label, 'bbox_xyxy': bbox_xyxy}
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def _load_image(self, path: Path) -> Tensor:
+        """Load an image as a C x H x W uint8 tensor.
```
Comment (nilsleh) [inline torchgeo/datasets/fmow.py]: we need to return float tensors for pytorch neural networks

File: tests/data/fmow/val/airport/airport_0/airport_0_0_rgb.json [hunk 3]
```diff
@@ -0,0 +1 @@
+{"bounding_boxes": [{"box": [1, 2, 3, 4]}]}
```
Comment (nilsleh) [inline tests/data/fmow/val/airport/airport_0/airport_0_0_rgb.json]: probably good to test with more than just one bounding box

**Review by nilsleh — CHANGES_REQUESTED** [review]
Can you also add a datamodule for this dataset. That should be quite straightforward. Additionally, regarding the download, we have several large datasets like SSL4EO that are also quite large but have the download option. So I would argue it is actually worthwhile to add here, or at least have documentation how to download that dataset for easier usability.

**Comment by adamjstewart** [comment]
Removed "Closes #2580" since this only covers the RGB version.

**Comment by nilsleh** [comment]
@arpitJ-dev Could you add a plotted sample example of the dataset somewhere for verification?

File: torchgeo/datasets/fmow.py (hunk 1 above)
Comment (arpitJ-dev) [reply torchgeo/datasets/fmow.py:236]: Added bounding-box rendering to `plot()`. I also tested it with an official fMoW validation image and added the sample plot to the PR.

File: tests/data/fmow/val/airport/airport_0/airport_0_0_rgb.json (hunk 3 above)
Comment (arpitJ-dev) [reply tests/data/fmow/val/airport/airport_0/airport_0_0_rgb.json]: Updated the test metadata to include multiple boxes and added coverage for the converted coordinates.

**Comment by arpitJ-dev** [comment]
@nilsleh Hey, sorry for the slow follow-up here. I actually added a plotted sample not long after you asked, it's the last section of the PR description above, right under the Testing checklist, titled "Sample plot". It's a real image from the official val split rather than the test fixtures. The title comes from sample['label'], and the red box is the ground truth annotation converted from the metadata's x/y/width/height into bbox_xyxy, and it lines up correctly around the runway and terminal.

I never came back to confirm it in a comment though, so let me know if that covers what you were looking for, or if you'd rather have it somewhere else too, like checked into the repo or generated from the test fixtures

**Review by nilsleh — APPROVED** [review]

**Comment by arpitJ-dev** [comment]
Hey @adamjstewart @robmarkcole, just flagging that this is green across the board now (20/20 checks, no conflicts) and nilsleh approved a few days back. Let me know if there's anything else needed on my end.

**Comment by adamjstewart** [comment]
I don't review AI-assisted PRs but hopefully someone else can review.

File: torchgeo/datamodules/fmow.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+"""Functional Map of the World datamodule."""
+
+from typing import Any
+
+from ..datasets import FMoW
+from .geo import NonGeoDataModule
+from .utils import collate_fn_detection
+
+
+class FMoWDataModule(NonGeoDataModule):
```
Comment (robmarkcole) [inline torchgeo/datamodules/fmow.py:13]: Can we apply normalisation like here, or better yet, any 'official' norm values? https://github.com/torchgeo/torchgeo/blob/481a8116c6844c410722fff463199b5fd37aa28d/torchgeo/datamodules/resisc45.py#L21-L23

File: torchgeo/datasets/fmow.py [hunk 5]
```diff
@@ -0,0 +1,287 @@
...
+        Args:
+            sample: a sample returned by __getitem__
+            show_titles: flag indicating whether to draw titles
+            suptitle: optional figure title
+
+        Returns:
+            Matplotlib Figure containing plotted sample
+        """
+        image = sample['image'].permute(1, 2, 0)
+        if image.is_floating_point() and image.max() > 1:
```
Comment (robmarkcole) [inline torchgeo/datasets/fmow.py:250]: I think this shouldn't be required if normalisation is applied

**Review by robmarkcole — CHANGES_REQUESTED** [review]
Overall looks good, could you take a look at the normalisation and default augs, thanks

**Comment by arpitJ-dev** [comment]
Added normalization to FMoWDataModule following the resisc45.py pattern. There's no published per-channel mean/std for fMoW-rgb (checked the official fMoW baseline repo and the SatMAE paper's fMoW appendix, neither has one), so I used the standard ImageNet RGB stats as a documented default. NonGeoDataModule picks these up automatically and applies K.Normalize to batches, same as resisc45.

On the max() > 1 check in the dataset's plot method: that one's on the raw sample from FMoW.__getitem__, before the datamodule's normalization ever touches it, since plotting works directly off dataset samples, not normalized batch tensors. So it's still doing its job (deciding whether to divide by 255 for display) independently of the datamodule change. Left it as is since removing it would break plotting on an unnormalized sample.

Let me know if you'd rather see it handled differently.

**Comment by nilsleh** [comment]
> Added normalization to FMoWDataModule following the resisc45.py pattern. There's no published per-channel mean/std for fMoW-rgb (checked the official fMoW baseline repo and the SatMAE paper's fMoW appendix, neither has one), so I used the standard ImageNet RGB stats as a documented default. NonGeoDataModule picks these up automatically and applies K.Normalize to batches, same as resisc45.
> 
> On the max() > 1 check in the dataset's plot method: that one's on the raw sample from FMoW.**getitem**, before the datamodule's normalization ever touches it, since plotting works directly off dataset samples, not normalized batch tensors. So it's still doing its job (deciding whether to divide by 255 for display) independently of the datamodule change. Left it as is since removing it would break plotting on an unnormalized sample.
> 
> Let me know if you'd rather see it handled differently.

@arpitJ-dev Please read the AI-POLICY, your communication on these review items is very clearly AI generated, which is a violation of the policy and I am therefore closing this PR.
